### PR TITLE
Fix TransactionPrograms Commit error handling.

### PR DESCRIPTION
**Pull request by pattyshack**
It is not safe to ignore txnPrograms' commit error for transactions that trigger programs invalidation.

Note: commit never returns error under sequential execution (except with bad test setup).

### CODE REVIEW

File: fvm/fvm.go [hunk 1]
```diff
@@ -141,7 +125,21 @@ func (vm *VirtualMachine) Run(
...
+		return err
+	}
+
+	// Note: it is safe to skip committing the parsed programs for non-normal
+	// transactions (i.e., bootstrap and script) since these do not invalidate
+	// programs.
+	if proc.Type() == TransactionProcedureType {
+		// NOTE: It is not safe to ignore txnPrograms' commit error for
+		// transactions that trigger programs invalidation.
+		return txnPrograms.Commit()
```
Comment (Tonix517) [inline fvm/fvm.go:139]: for understanding purpose, how should we handle the error from `Commit()` here, is the current error handling enough for this new error source?

**Review by Tonix517 — APPROVED** [review]

File: fvm/fvm.go (hunk 1 above)
Comment (pattyshack) [reply fvm/fvm.go:139]: this will halt the execution node.

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors retry

**Pull request merged**